### PR TITLE
refactor(ui): change record button

### PR DESCRIPTION
**Pull request by Matrixbirds**
optimize recording button

### CODE REVIEW

**Review by Cheerego7 — DISMISSED** [review]

File: desktop/renderer-app/src/pages/BigClassPage/index.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
     TopBar,
     TopBarDivider,
     LoadingPage,
+    CloudRecordBtn,
```
Comment (Cheerego7) [inline desktop/renderer-app/src/pages/BigClassPage/index.tsx:11]: remove unused the RecordButton component

**Review by Cheerego7 — DISMISSED** [review]

**Comment by LitoMore** [comment]
SVGs are not compressed properly.

Please rebase the main branch and rerun `yarn run bootstrap` to make sure dependencies are up to date.

**Review by LitoMore — CHANGES_REQUESTED** [review]
SVGs are not compressed properly.

Please rebase with the main branch and rerun `yarn run bootstrap` to make sure dependencies are up to date.

File: web/flat-web/src/pages/SmallClassPage/index.tsx [hunk 2]
```diff
@@ -240,23 +240,7 @@ export const SmallClassPage = observer<SmallClassPageProps>(function SmallClassP
...
-                                onStop() {
-                                    void message.success(t("recording-completed-tips"));
-                                },
-                            })
-                        }
-                    />
-                )}
-            </>
-        );
+        return <>{renderClassMode()}</>;
```
Comment (BlackHole1) [inline web/flat-web/src/pages/SmallClassPage/index.tsx]:
```suggestion
        return renderClassMode();
```

File: web/flat-web/src/pages/OneToOnePage/index.tsx [hunk 3]
```diff
@@ -170,25 +170,7 @@ export const OneToOnePage = observer<OneToOnePageProps>(function OneToOnePage()
...
-                                onStop() {
-                                    void message.success(t("recording-completed-tips"));
-                                },
-                            })
-                        }
-                    />
-                )}
-            </>
-        );
+        return null;
```
Comment (BlackHole1) [inline web/flat-web/src/pages/OneToOnePage/index.tsx]: why don't remove renderTopBarCenter function?

File: web/flat-web/src/stores/class-room-store.ts [hunk 4]
```diff
@@ -183,6 +185,15 @@ export class ClassRoomStore {
...
         );
 
+        reaction(
+            () => this.isRecording,
+            (isRecording: boolean) => {
+                if (isRecording) {
+                    void message.success(i18next.t("start-recording"));
+                }
+            },
+        );
```
Comment (BlackHole1) [inline web/flat-web/src/stores/class-room-store.ts:195]: @crimx Can we write it that way? Would it be better to put it in a component

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: packages/flat-components/src/components/ClassroomPage/CloudRecordBtn/index.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export type CloudRecordBtnProps = {
+    isRecording: boolean;
+    onClick: (evt: any) => void;
+};
+
+export const CloudRecordBtn: React.FC<CloudRecordBtnProps> = observer(
+    ({ isRecording, onClick }) => {
+        return (
+            <TopBarRightBtn
+                title={isRecording ? "Stop Recording" : "Start Recording"}
```
Comment (BlackHole1) [inline packages/flat-components/src/components/ClassroomPage/CloudRecordBtn/index.tsx]: i18n

File: packages/flat-components/src/index.ts [hunk 6]
```diff
@@ -19,3 +19,4 @@ export * from "./components/RoomDetailPage";
 export * from "./components/RoomStatusElement";
 export * from "./containers/CloudStorageContainer";
 export * from "./components/DeviceTestPage";
+export * from "./components/ClassroomPage/CloudRecordBtn";
```
Comment (BlackHole1) [inline packages/flat-components/src/index.ts]:
```suggestion
export * from "./components/ClassroomPage/CloudRecordBtn";

```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

File: desktop/renderer-app/src/pages/SmallClassPage/index.tsx [hunk 7]
```diff
@@ -259,23 +259,7 @@ export const SmallClassPage = observer<SmallClassPageProps>(function SmallClassP
...
-                                onStop() {
-                                    void message.success(t("recording-completed-tips"));
-                                },
-                            })
-                        }
-                    />
-                )}
-            </>
-        );
+        return <>{renderClassMode()}</>;
```
Comment (BlackHole1) [inline desktop/renderer-app/src/pages/SmallClassPage/index.tsx]:
```suggestion
        return renderClassMode();
```

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

**Review by BlackHole1 — APPROVED** [review]

**Comment by BlackHole1** [comment]
please re-review
cc: @LitoMore

**Review by LitoMore — APPROVED** [review]

**Pull request merged**